tt_helpers.go

```diff
@@ -0,0 +1,121 @@
+
+const (
+	kType       = "type"
+	kIsNullable = "is_nullable"
+)
+
+const (
+	defaultIgnoreChars   = ""
+	defaultDecSeparators = "."
+	defaultNullValue     = ""
+)
+
+// MappersTable can return a Mapper[string] based on the type name.
+type MappersTable interface {
+	getMapper(typ TypeName) (Mapper[string], bool)
+}
+
+// ParsersTTTable is basic tarantool parsers table.
+type ParsersTTTable struct{}
+
+// getMapper is the implementation of MappersTable for MappersTTTable.
+func (ptable *ParsersTTTable) getMapper(typ TypeName) (Mapper[string], bool) {
+	mapper, ok := parserByTTType[typ]
+	return mapper, ok
+}
+
+// parserByTTType is a mapping from tarantool typenames to parsers.
+var parserByTTType = map[TypeName]Mapper[string]{
+	TypeBoolean:   NewBoolParser(),
+	TypeString:    NewStringParser(),
+	TypeUnsigned:  NewUIntParser(defaultIgnoreChars),
+	TypeDatetime:  NewDateTimeParser(),
+	TypeUUID:      NewUUIDParser(),
+	TypeMap:       NewMapParser(),
+	TypeArray:     NewSliceParser(),
+	TypeVarbinary: NewBinaryParser(),
+
+	TypeDouble:  NewFloatParser(defaultIgnoreChars, defaultDecSeparators),
+	TypeDecimal: NewDecimalParser(defaultIgnoreChars, defaultDecSeparators),
+
+	TypeInteger: MakeSequenceMapper([]Mapper[string]{
+		NewUIntParser(defaultIgnoreChars),
+		NewIntParser(defaultIgnoreChars),
+	}),
+
+	TypeNumber: MakeSequenceMapper([]Mapper[string]{
+		NewUIntParser(defaultIgnoreChars),
+		NewIntParser(defaultIgnoreChars),
+		NewFloatParser(defaultIgnoreChars, defaultDecSeparators),
+	}),
+
+	TypeAny: MakeSequenceMapper([]Mapper[string]{
+		NewUIntParser(defaultIgnoreChars),
+		NewIntParser(defaultIgnoreChars),
+		NewFloatParser(defaultIgnoreChars, defaultDecSeparators),
+		NewDecimalParser(defaultIgnoreChars, defaultDecSeparators),
+		NewBoolParser(),
+		NewDateTimeParser(),
+		NewIntervalParser(),
+		NewUUIDParser(),
+		NewStringParser(), // Always success.
+	}),
+
+	// Same as any by default.
+	TypeScalar: MakeSequenceMapper([]Mapper[string]{
+		NewUIntParser(defaultIgnoreChars),
+		NewIntParser(defaultIgnoreChars),
+		NewFloatParser(defaultIgnoreChars, defaultDecSeparators),
+		NewDecimalParser(defaultIgnoreChars, defaultDecSeparators),
+		NewBoolParser(),
+		NewDateTimeParser(),
+		NewIntervalParser(),
+		NewUUIDParser(),
+		NewStringParser(), // Always success.
+	}),
+
+	TypeInterval: NewIntervalParser(),
+}
+
+var ttParsers MappersTable = &ParsersTTTable{}
+
+// NewParserFor returns parser by the type name and `isNullable` property.
+var NewParserFor = func(typ TypeName, isNullable bool) (Mapper[string], error) {
+	parser, isSupported := ttParsers.getMapper(typ)
+	if !isSupported {
+		return nil, NewErrUnexpectedType(typ)
+	}
+	if isNullable {
+		parser = MakeSequenceMapper([]Mapper[string]{
+			parser,
+			NewNullParser(defaultNullValue),
+		})
+	}
+	return parser, nil
+}
+
+// GetStringsToTTFmt creates Mapper list by the space format.
+func GetStringsToTTFmt(spaceFmt []map[string]any) ([]Mapper[string], error) {
```

Comment:
I would not like to see `map[any]any` in the public interface. We could add helper structures for CallTyped().

```suggestion
// GetStringsToTTFmt creates Mapper list by the space format.
func GetStringsToTTFmt(spaceFmt []map[string]any) ([]Mapper[string], error) {
```

Reply:
Here no any `map[any]any`, or what does it mean?

Reply:
Oh, sorry `map[string]any` - it is not critical. The idea is that a space field format could be defined as a GoLang struct, not just `map[string]any`:

https://github.com/tarantool/go-tarantool/blob/b30e2b69c9fca74953178afe9a80c38530d85aff/schema.go#L154-L160

Reply:
Ok, changed. Since https://github.com/tarantool/go-tarantool/commit/40fe0d3a6926559328bf4e1cd93d4c90a5b316a4 was not included in the release, the master revision of `go-tarantool` used as dependency.